packages/components/inline-edit/inline-edit.ts

```diff
@@ -0,0 +1,375 @@
+
+    /** @docs-private */
+    protected readonly placements = PopUpPlacements;
+
+    /** @docs-private */
+    protected readonly colors = KbqComponentColors;
+
+    /** @docs-private */
+    private readonly elementRef = inject(ElementRef);
+
+    private initialValue: unknown;
+
+    constructor() {
+        toObservable(this.mode)
+            .pipe(skip(1), takeUntilDestroyed())
+            .subscribe((currentMode) => this.modeChange.emit(currentMode));
+    }
+
+    /** @docs-private */
+    protected toggleMode(): void {
+        this.mode.update((mode) => (mode === 'view' ? 'edit' : 'view'));
+    }
+
+    /** @docs-private */
+    protected onClick(event: Event): void {
+        if (this.disabled() || this.isEditMode()) return;
+
+        event.stopPropagation();
+        this.toggleMode();
+    }
+
+    /** @docs-private */
+    protected onAttach(): void {
+        this.setOverlayWidth();
+        const formFieldRef = this.formFieldRef();
+
+        formFieldRef?.control.stateChanges
+            .pipe(takeUntil(this.overlayDir()!.overlayRef.detachments()))
+            .subscribe(() => {
+                if (!this.isInvalid()) {
+                    const tooltipTrigger = this.tooltipTrigger();
+
+                    tooltipTrigger.isOpen && tooltipTrigger.hide();
+                }
+            });
+
+        setTimeout(() => {
+            formFieldRef?.focus();
+            this.initialValue = this.getValue();
+
+            const input: HTMLInputElement | HTMLTextAreaElement | null =
+                this.overlayDir()!.overlayRef.overlayElement.querySelector('input,textarea');
+
+            if (this.initialValue) input?.select();
+        }, 0);
+    }
+
+    /** @docs-private */
+    protected save($event?: Event): void {
+        $event?.stopPropagation();
+
+        if (this.isInvalid()) {
+            this.showTooltipOnError() && this.tooltipTrigger()?.show();
+        } else {
+            this.toggleMode();
+            this.saved.emit();
+        }
+    }
+
+    /** @docs-private */
+    protected cancel(): void {
+        this.setValue(this.initialValue);
+
+        this.toggleMode();
+        this.canceled.emit();
+    }
+
+    /** @docs-private */
+    protected onOverlayKeydown(event: KeyboardEvent): void {
+        const { target, key } = event;
+
+        switch (key) {
+            // @TODO Handle Tab keydown (#DS-4160)
+            case 'Escape': {
+                this.cancel();
+                break;
+            }
+            case 'Enter': {
+                if (hasModifierKey(event, 'ctrlKey', 'metaKey') || !(target instanceof HTMLTextAreaElement)) {
+                    this.save();
+                }
+
+                break;
+            }
+            default: {
+                return;
+            }
+        }
+    }
+
+    private isInvalid(): boolean {
+        const formFieldRef = this.formFieldRef();
+
+        if (!formFieldRef) return false;
+
+        return formFieldRef.control.errorState;
```

Comment:
у ff есть свойство invalid
```suggestion
        return formFieldRef.invalid;
```